src/components/NewResponsibleModal/hook.ts

```diff
@@ -1,19 +1,35 @@
+import { useState } from "react";
+
+import { useQuery } from "@tanstack/react-query";
 import { useSearchParams } from "react-router";
 
+
 import type { ResponsibleData } from "./interface";
 import { strings } from "../../constants";
 import { useFilters } from "../../hooks/useFilters";
 import { useToast } from "../../hooks/useToast";
+import { fetchAddress, type Address } from "../../services/address";
+import { createFamilyMember, createFamilyMemberAddress } from "../../services/family-members";
 import { useDateInput } from "../Inputs/DateInput/hook";
 import { useSelectInput } from "../Inputs/SelectInput/hook";
 import { useTextInput } from "../Inputs/TextInput/hook";
 
-export function useNewResponsibleModal() {
+export function useNewResponsibleModal(studentId?: string) {
```

Comment:
Por que studentId é opcional? Não podemos adicionar um responsável sem estudante.
É possível deixar obrigatório?
```suggestion
export function useNewResponsibleModal(studentId: string) {
```